sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java

```diff
@@ -101,6 +103,35 @@ public DDLStatementDispatcher(BlobIndices blobIndices,
         }
 
         @Override
+        public ListenableFuture<Long> visitOptimizeTableStatement(OptimizeTableAnalyzedStatement analysis, UUID jobId) {
+            if (analysis.indexNames().isEmpty()) {
+                return Futures.immediateFuture(null);
+            }
+            ForceMergeRequest request = new ForceMergeRequest(analysis.indexNames().toArray(
+                new String[analysis.indexNames().size()]));
+
+            // Pass parameters to ES request
+            Integer maxNumSegments = analysis.settings().getAsInt(OptimizeSettings.MAX_NUM_SEGMENTS.name(), null);
```

Comment:
either make fully use of the crate setting logic, or just use ES setting names here instead (I'd prefer 2nd). same for all other settings below